crates/solver/src/settlement_access_list.rs

```diff
@@ -76,16 +100,63 @@ impl TenderlyApi {
     }
 }
 
+#[async_trait::async_trait]
+impl AccessListEstimating for TenderlyApi {
+    async fn estimate_access_lists(
+        &self,
+        web3: &Web3,
+        network_id: String,
+        txs: &[TransactionBuilder<DynTransport>],
+    ) -> Vec<Result<AccessList>> {
+        futures::future::join_all(txs.iter().map(|tx| async {
+            let input = tx
+                .data
+                .clone()
+                .ok_or(anyhow!("transaction data does not exist"))?;
+            let from = tx
+                .from
+                .clone()
+                .ok_or(anyhow!("transaction from does not exist"))?
+                .address();
+            let to = tx.to.ok_or(anyhow!("transaction to does not exist"))?;
+            let block_number = web3.eth().block_number().await?.as_u64();
```

Comment:
Would it be problematic if the current block changed during the execution of `estimate_access_lists()`?
Does it make sense to fetch the `block_number` once and reuse it for all the futures?

Reply:
Nice catch. Here, I was going for a safe option (delay the `block_number` call to the last minute possible). Theoretically, with each new block access list could change (and also our submission logic last 120s which means a lot of blocks could be mined), but practically, and given our app specific needs, it is possible that this is overkill. Notified this, so I can check later when we build the metrics.

Reply:
Also we can use Tenderly's view of the what the latest block is:
```
```

Additionally, at least for the `api/.../simulate` (which you can observe in the developer tools when simulating a transaction in their UI) they use `block_number: null` to indicate simulation on the pending block (which is probably what we want right?) and it includes the actual block that was used in the response. It might also work for the access list API?

<img width="661" alt="image" src="https://user-images.githubusercontent.com/4210206/155985040-54e4709a-52b4-401f-89d0-adae7b67e272.png">

Reply:
Tried with null, not working. Added the `block_number` query.